portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,19 +37,51 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-    
-  private List<String> names;
- 
-  @Override
-  public void init() {
-    names = new ArrayList<String>(Arrays.asList("Sam", "Hadley", "Jerry"));
-  }
+  
+  /** Java object converter. */
+  private static final Gson gson = new Gson();
+
+  /** The max amount of comments that will be displayed. */
+  private int maxComments = 5;
+
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String name = names.get((int) (Math.random() * names.size()));
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello " + name + "!");
+    Query query = new Query("Comment").addSort("timeStamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    List<Entity> resultsList = results.asList(FetchOptions.Builder.withLimit(maxComments));
 
+    List<Comment> comments = new ArrayList<>();
```

Comment:
Consider moving this logic for fetching the List of Comments to its own function. This would help make this function easier to read.

Reply:
Not sure what you mean.